src/internal/m365/collection/drive/collections_tree.go

```diff
@@ -335,19 +339,17 @@ func (c *Collections) populateTree(
 				el.AddRecoverable(ctx, clues.Stack(err))
 			}
 
-			counter.Inc(count.PagesEnumerated)
+			pageCount++
+
+			pageItemCount += len(page)
 
 			// Stop enumeration early if we've reached the page limit. Keep this
 			// at the end of the loop so we don't request another page (pager.NextPage)
 			// before seeing we've passed the limit.
-			if limiter.hitPageLimit(int(counter.Get(count.PagesEnumerated))) {
+			if limiter.hitPageLimit(pageCount) {
```

Comment:
I made a comment in one of the other PRs, but we may actually want to keep this as the total number of pages we've seen? It's a bit fuzzy either way, but I guess I'd err on the side of stopping processing earlier during preview backups since we don't expect many resets but may have multiple delta enumerations (which also reset the page count checked here)

Reply:
It's currently buggy either way.  Other comment linked the fix using a counter, so I'll  keep that in the later PR.